### PR TITLE
XW-514 Convert routes to ES6

### PR DESCRIPTION
**Pull request by rogatty**
https://wikia-inc.atlassian.net/browse/XW-514

### CODE REVIEW

File: front/scripts/main/routes/ApplicationRoute.js [hunk 1]
```diff
@@ -58,15 +59,16 @@ App.ApplicationRoute = Em.Route.extend(
...
-					title: string,
-					trackingCategory: string,
-					info: LinkInfo,
-					// exec() returns an array of matches or null if no match is found.
-					domainNameRegExpMatchArray: RegExpExecArray = /\.[a-z0-9\-]+\.[a-z0-9]{2,}$/i.exec(window.location.hostname),
-					cookieDomain: string = domainNameRegExpMatchArray ? '; domain=' + domainNameRegExpMatchArray[0] : '',
-					defaultSkin: string = Em.getWithDefault(Mercury, 'wiki.defaultSkin', 'oasis');
+			handleLink(target) {
+				const currentRoute = this.router.get('currentRouteName'),
+				// exec() returns an array of matches or null if no match is found.
```
Comment (hakubo) [inline front/scripts/main/routes/ApplicationRoute.js]: missing tab?

File: front/scripts/main/routes/ApplicationRoute.js [hunk 2]
```diff
@@ -58,15 +59,16 @@ App.ApplicationRoute = Em.Route.extend(
...
+					domainNameRegExpMatchArray = (/\.[a-z0-9\-]+\.[a-z0-9]{2,}$/i).exec(window.location.hostname),
+					cookieDomain = domainNameRegExpMatchArray ? `; domain=${domainNameRegExpMatchArray[0]}` : '',
+					defaultSkin = Em.getWithDefault(Mercury, 'wiki.defaultSkin', 'oasis');
+
+				let	title,
```
Comment (hakubo) [inline front/scripts/main/routes/ApplicationRoute.js]: one tab too many? :)

**Comment by hakubo** [comment]
LGTM

File: front/scripts/main/routes/ApplicationRoute.js (hunk 1 above)
Comment (rogatty) [inline front/scripts/main/routes/ApplicationRoute.js]: Autoformatting in WebStorm doesn't like those inline comments and I had to fix it manually every time :/

@hakubo what do you think about replacing them with JSDoc?
```
const currentRoute = this.router.get('currentRouteName'),
	/**
	 * exec() returns an array of matches or null if no match is found.
	 */
	domainNameRegExpMatchArray = (/\.[a-z0-9\-]+\.[a-z0-9]{2,}$/i).exec(window.location.hostname),
```

File: front/scripts/main/routes/ApplicationRoute.js [hunk 3]
```diff
@@ -202,15 +209,16 @@ App.ApplicationRoute = Em.Route.extend(
...
+			let adsInstance;
 
-			if (M.prop('adsUrl') && !M.prop('queryParams.noexternals') && !instantGlobals.wgSitewideDisableAdsOnMercury) {
+			if (M.prop('adsUrl') && !M.prop('queryParams.noexternals') &&
+				!instantGlobals.wgSitewideDisableAdsOnMercury) {
 				adsInstance = Mercury.Modules.Ads.getInstance();
 				adsInstance.init(M.prop('adsUrl'));
 
-				/**
+				/*
```
Comment (tomasznapieralski) [inline front/scripts/main/routes/ApplicationRoute.js]: missing `*`

File: front/scripts/main/routes/ApplicationRoute.js (hunk 1 above)
Comment (Warkot) [inline front/scripts/main/routes/ApplicationRoute.js]: Shouldn't `exec()` already have the documentation in the standard lib?

**Comment by tomasznapieralski** [comment]
lgtm

**Pull request merged**